enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,504 @@
+* Allow aggregating portions of managed service metrics at a multi-cluster level, e.g. for reporting SLOs.
+* Allow longer retention for portions of data (e.g. calculating 1 month burn down SLO budget).
+* Allow analytics on portion of managed service data in conjuction with Telemeter data at the aggregation level.
+* End customers using UWM should not see managed service metrics.
+* Soft Goal: A foundation for next iterations mentioned in ### Long term motivation.
+
+### Long Term Motivation
+
+To avoid maintaining multiple monitoring solutions, this proposal aims to be a foundation for future monitoring enhancements. Therefore, the following goals are left out of scope for the current enhancement but are kept in mind for future iterations:
+
+* Such a solution as a community project which is easy to use & contribute to.
+* Reduce Platform Monitoring overhead and its single point of failure (SPOF) property.
+* Auto-scaling metric scraping based on number of ServiceMonitors
+* Soft multi-tenancy for local storage (TSDB).
+* Opt-in for full forwarding mode.
+* Full architecture for Hypershift ([WIP](https://docs.google.com/document/d/1BJarERppgiJ8esc6d8anbJMOQ0AflFBQea-Zc9wAp0s/edit))
+
+### Non-Goals
+
+* Access controls for managed service owners for metrics from different managed services. For example, there are currently no requirements for owners of managed services A to be prohibited from seeing metrics from managed service B, and vice versa.
+* Expose managed service metrics to end-users. We know e.g. OCS addon wants this, but this can be achieved on top of the proposed solution using client or backend side filtering for data isolation if needed (e.g. prom-label-proxy). The monitoring team can do this in the next iteration.
+* Integrate managed service metrics into the OpenShift console. We know OCS wants to integrate with OCP Console, but we assume the console team can help support switching between Prometheus-compatible data sources.
+* Solve Platform Monitoring SPOF, scalability and overhead. This will be targeted in a follow-up enhancement.
+* Allow logging and tracing capabilities. Other solutions should tackle this (designed by Logging and Tracing teams.)
+
+## Proposal
+
+### Personas
+
+* OSD/Platform SRE
+* MTSRE
+* RHOBS Devs
+* MTO
+* Customer: User of Managed Service and OSD
+
+### User Stories
+
+#### Story 1
+
+As a MTSRE/MTO, I would like to define ServiceMonitors that will scrape metrics from my managed service by a Prometheus-compatible system.
+
+#### Story 2
+
+As an MTSRE/MTO, I would like to query SaaS/Addon metrics using PromQL to understand the health or troubleshoot the performance of my managed service.
+
+#### Story 3
+
+As an MTSRE/MTO, I would like to create alerting and recording rules for Prometheus metrics exposed by my managed service.
+
+#### Story 4
+
+As an MTSRE, I would like to configure routing for alerts defined on Prometheus metrics exposed by my SaaS/Addon.
```

Comment:
Again proposing to abstract routing away, in favor of PrometheusServiceLevel CR or something similar (sloth):

> As an MTSRE/MTO, I would like to create SLO-oriented CRs, that will configure alerts defined on Prometheus metrics exposed by my SaaS/Addon.